zvariant/src/type.rs

```diff
@@ -433,42 +433,54 @@ macro_rules! map_impl {
 map_impl!(BTreeMap<K: Ord, V>);
 map_impl!(HashMap<K: Eq + Hash, V, H: BuildHasher>);
 
-impl Type for Duration {
-    fn signature() -> Signature<'static> {
-        <(u64, u32)>::signature()
+impl_type! {
+    Duration => (u64, u32) {
+        test_duration {
```

Comment:
Let's remove `test_` prefix from test mod names please. It's redundant given the context.